src/raw/Retrieve.fs

```diff
@@ -0,0 +1,162 @@
+// Copyright 2019 Cognite AS
+// SPDX-License-Identifier: Apache-2.0
+
+namespace CogniteSdk.Raw
+
+open System.Net.Http
+open System.Runtime.CompilerServices
+open System.Runtime.InteropServices
+open System.Threading
+open System.Threading.Tasks
+
+open FSharp.Control.Tasks.V2.ContextInsensitive
+open Oryx
+open Thoth.Json.Net
+
+open CogniteSdk
+
+[<AutoOpen>]
+module Retrieve =
+    [<Literal>]
+    let Url = "/raw/dbs"
+
+    type DatabaseResponse = {
+        Items: DatabaseReadDto seq
+    } with
+         static member Decoder : Decoder<DatabaseResponse> =
+            Decode.object (fun get -> {
+                Items = get.Required.Field "items" (Decode.list DatabaseReadDto.Decoder |> Decode.map seq)
+            })
+
+    type TableResponse = {
+        Items: TableReadDto seq
+    } with
+         static member Decoder : Decoder<TableResponse> =
+            Decode.object (fun get -> {
+                Items = get.Required.Field "items" (Decode.list TableReadDto.Decoder |> Decode.map seq)
+            })
+
+    let tablesCore (database: string) (queryParameters: DatabaseQuery seq) (fetch: HttpHandler<HttpResponseMessage, 'a>) =
+        let decodeResponse = Decode.decodeResponse TableResponse.Decoder (fun response -> response.Items)
+        let queries = queryParameters |> Seq.map DatabaseQuery.Render |> List.ofSeq
+
+        GET
+        >=> setVersion V10
+        >=> setResource (Url + "/" + database + "/tables")
```

Comment:
We need to check if database and table names can be e.g utf-8 endoded. If so we need to URL encode the entries in the URLs.

Reply:
Have tested. We must URL encode. For an example, see https://github.com/cognitedata/cognite-odata-fusion/blob/master/src/Controllers/LegacyOAuthController.fs#L26